File: app/src/main/java/com/hotels/tajawal/dubaihotels/hotels/data/HotelDataSource.java
package com.hotels.tajawal.dubaihotels.hotels.data;

import com.hotels.tajawal.dubaihotels.base.BaseObservable;
import com.hotels.tajawal.dubaihotels.hotels.model.HotelResponse;
import com.hotels.tajawal.dubaihotels.remote.RetrofitProvider;

import io.reactivex.Observable;
import retrofit2.Retrofit;

/**
 * Hotel Data source
 * Created by Sara Elmoghazy.
 */
public class HotelDataSource extends BaseObservable<HotelResponse> {

    RetrofitProvider retrofitProvider;
    HotelService hotelService;

    public HotelDataSource(RetrofitProvider retrofitProvider) {
        this.retrofitProvider = retrofitProvider;
        hotelService = retrofitProvider.getRetrofit().create(HotelService.class);
    }


    public Observable<HotelResponse> getHotels() {
        return getObservable(hotelService.getHotels());

    }
}

File: app/src/main/java/com/hotels/tajawal/dubaihotels/hotels/data/HotelDataSource.kt
//package com.hotels.tajawal.dubaihotels.hotels.data
//
//import com.hotels.tajawal.dubaihotels.base.BaseObservable
//import com.hotels.tajawal.dubaihotels.hotels.model.HotelResponse
//import io.reactivex.Observable
//import retrofit2.Retrofit
//import javax.inject.Inject
//
///**
// * Created by Sara Elmoghazy on 19,November,2018
// */
//class HotelDataSource @Inject constructor(
//        private val retrofit: Retrofit) : BaseObservable<HotelResponse>() {
//
//    fun getHotels(): Observable<HotelResponse> {
//        return retrofit.create(HotelService::class.java).getHotels()
//    }
//
//}